base_geoengine/static/src/js/mapfish/GeoStat.js

```diff
@@ -399,14 +399,15 @@ mapfish.GeoStat.Distribution = OpenLayers.Class({
     classifyByQuantils: function(nbBins) {
         var values = this.values;
         values.sort(function(a,b) {return a-b;});
-        var binSize = Math.round(this.values.length / nbBins);
+        // We don't want more classes that elements to classify
+        var maxNbBins = Math.min(nbBins, values.length - 1);
```

Comment:
> Sorry, I got confused with Python. Very late review...

That's why multi reviewers is always a good practice. We are all humans.. :smirk: